api/v1alpha08/sonataflow_types.go

```diff
@@ -24,6 +24,535 @@ import (
 	"github.com/kiegroup/kogito-serverless-operator/api"
 )
 
+const DefaultContainerName = "workflow"
+
+// FlowContainer is the container for the internal workflow deployment based on the default Kubernetes Container API
+type FlowContainer struct {
+	// Container image name.
+	// More info: https://kubernetes.io/docs/concepts/containers/images
+	// This field is optional to allow higher level config management to default or override
+	// container images in workload controllers like Deployments and StatefulSets.
+	// +optional
+	Image string `json:"image,omitempty" protobuf:"bytes,2,opt,name=image"`
```

Comment:
Another silly question. So, if the user configures an image here, the operator will basically skip the image generation for that workflow, and just grab the configured one, right?
Which use case are we trying to cover with this? 
I mean, looks like a very advanced situation where the build was produced completely outside the operator scope, no warranty that the image even corresponds to a workflow, so this why I'm asking, I which situations people might want to do that? optimizations maybe?

Reply:
Hey @wmedvede thanks for the question. Please see [https://issues.redhat.com/browse/KOGITO-9833](KOGITO-9833) where I summarize the fix for this problem. This is the first question one will have once it tries this feature. The operator uses the `spec.flow` to do any configurations like discovery and eventing objects creation. So if one sets an image that is not compatible, it happens to have a bad deployment in the end. But we will do an integrity check before deploying it in the future.